src/ngx_http_lua_shdict.c

```diff
@@ -1823,9 +1957,35 @@ ngx_http_lua_ffi_shdict_incr(ngx_shm_zone_t *zone, u_char *key,
     dd("shdict lookup returned %d", (int) rc);
 
     if (rc == NGX_DECLINED || rc == NGX_DONE) {
-        ngx_shmtx_unlock(&ctx->shpool->mutex);
-        *err = "not found";
-        return NGX_ERROR;
+        if (arg_num == 0) {
+            ngx_shmtx_unlock(&ctx->shpool->mutex);
+            *err = "not found";
+            return NGX_ERROR;
+        }
+
+        /* add value */
+        num = *value + init;
+
+        if (rc == NGX_DONE) {
+
+            if ((size_t) sd->value_len == sizeof(double)) {
```

Comment:
I think we should check the value type instead of simply checking value length. This is because different value types MAY have identical value lengths.